flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/rules/physical/batch/BatchPhysicalHashJoinRule.scala

```diff
@@ -50,23 +47,10 @@ class BatchPhysicalHashJoinRule
 
   override def matches(call: RelOptRuleCall): Boolean = {
     val join: Join = call.rel(0)
-    val joinInfo = join.analyzeCondition
-    // join keys must not be empty
-    if (joinInfo.pairs().isEmpty) {
-      return false
-    }
-
     val tableConfig = unwrapTableConfig(call)
-    val isShuffleHashJoinEnabled = !isOperatorDisabled(tableConfig, OperatorType.ShuffleHashJoin)
-    val isBroadcastHashJoinEnabled =
-      !isOperatorDisabled(tableConfig, OperatorType.BroadcastHashJoin)
-
-    val leftSize = binaryRowRelNodeSize(join.getLeft)
-    val rightSize = binaryRowRelNodeSize(join.getRight)
-    val (isBroadcast, _) = canBroadcast(join.getJoinType, leftSize, rightSize, tableConfig)
 
-    // TODO use shuffle hash join if isBroadcast is true and isBroadcastHashJoinEnabled is false ?
-    if (isBroadcast) isBroadcastHashJoinEnabled else isShuffleHashJoinEnabled
+    canUseJoinStrategy(join, tableConfig, JoinStrategy.BROADCAST) ||
```

Comment:
we can get TableConfig from RelNode using unwrapTableConfig(join). So tableConfig is not necessary parameter